### ISSUE TITLE
Professor Putricide and Malleable Goo

### ISSUE THREAD

**Issue by Guzzone** [dalaranwow/dalaran-wow#5878]
[//]: # (INFORMATION ABOUT TAGS)
[//]: # (The black and yellow tag is unique. There can only be one label assigned to them - one black and one yellow)
[//]: # (Priority - ... tags are assigned by team members)
[//]: # (Your issue will be reviewed by testers and the developers will fix it in the future.)
[//]: # (Only once the issue got the label [Fixed - On Live Server] you can find the fix on your Gameserver.)
[//]: # (You can increase the relevance of your issue by adding a thumb-up emote http://i.imgur.com/RFouVVi.png)

[//]: # (***************************************************************)
[//]: # (** !!!!!!!!!!!!!DON'T DELETE THIS TEMPLATE ELSE YOUR ISSUE WILL BE CLOSED!!!!!!!!!!!! **)
[//]: # (***************************************************************)

I'm including all of the issues in one thread as they might be, and likely are, related

**Current behaviour**: Professor Putricide's spell cast of Malleable Goo does not appear in the combat log, this applies to every difficulty.
The boss emote is something like: "|TInterface\Icons\inv_misc_herb_evergreenmoss.blp:16|t%s cast |cFF00FF00Malleable Goo!|r Professor Putricide cast Malleable Goo!".
"|TInterface\Icons\inv_misc_herb_evergreenmoss.blp:16|t%s cast |cFF00FF00Malleable Goo!|r" is the adders' tongue icon, which is also the Malleable Goo's debuff icon.

**Expected behaviour**:
PP's cast of Malleable Goo (with target) should appear in the combat log.
The boss emote should be: "Professor Putricide cast Malleable Goo!" like it is "Professor Putricide cast Choking Gas Bomb"

**Steps to reproduce the problem**:
Get Professor Putricide to phase 2 and wait for him to cast Malleable Goo.
You will only see "Malleable Goo" entries in the log when:

-  it actually hits someone (SPELL_DAMAGE)
-  it misses someone (SPELL_MISSED)
-  it applies the haste debuff to someone (SPELL_AURA_APPLIED) 
-  the haste debuff runs out (SPELL_AURA_REMOVED)

You will not see any entry when the boss casts the spell (SPELL_CAST_SUCCESS).

**Include proofs for this behaviour**
https://www.dropbox.com/s/ted7f1nazr3pn4x/pp2.zip
This file (log from last night) includes several fights on Professor Putricide 25 heroic difficulty but this applies to all difficulties as far as I'm aware.
If you don't want to read a barely understandable file with several hundreds of thousands of lines you can take my word for it. If you instead feel like it you can search Malleable Goo inside the file and you won't find a single instance of the spell cast being in the log.

DBM's putricide.lua has a function to start timers
```
function mod:SPELL_CAST_SUCCESS(args)
	[...]
	elseif args:IsSpellID(72615, 72295, 74280, 74281) then
		warnMalleableGoo:Show()
		specWarnMalleableGooCast:Show()
		if mod:IsDifficulty("heroic10") or mod:IsDifficulty("heroic25") then
			timerMalleableGooCD:Start(20)
		else
			timerMalleableGooCD:Start()
		end
		if self.Options.BypassLatencyCheck then
			self:ScheduleMethod(0.1, "OldMalleableGooTarget")
		else
			self:ScheduleMethod(0.1, "MalleableGooTarget")
		end
	end
```
assuming this used to work on retail, it should work here.

**Include the ID for the game objects, npcs (creatures,pets,minions), spells, items, quests, instances, zones, achievements, skills**
SpellIDs should be:

-  25nm: 72615
-  10nm: 72295
-  10hc: 74280
-  25hc: 74281

**Include Screenshots from the issue if necessary**
https://www.youtube.com/watch?v=2SuZyN5b5uY
If you go to 5:05 in this video you can clearly see the difference between DBM announcements and emotes. There is an emote for Choking Gas Bombs as there is a DBM announcement, there is an emote for Mallleable Goo but there is no DBM announcement

**Comment by Guzzone** [comment]
I'm not sure how I should've formatted the title, let me know if I should edit it in any way

**Comment by Guzzone** [comment]
I'm almost sure what I said about the icon is wrong someway, afaik add-ons that read the emote don't work though.
I'll update as soon as I understand what the issue is, the part about dbm and the log is correct

**Comment by Guzzone** [comment]
It seems that BigWigs also reads the combat log to announce and start goo timers

`self:Log("SPELL_CAST_SUCCESS", "BouncingGooBall", 72295, 74280, 72615, 74281) --10/25`

**Comment by Guzzone** [comment]
I've edited the post to remove the wrong informations about the emote's icon which, from watching retail videos, works as intended.
I've changed the file to include a bit of the Festergut's 25hc fight as the announcement doesn't work there either.

**Comment by Guzzone** [comment]
Festergut on BigWigs
```
do
	local goo = GetSpellInfo(72310)
	function mod:UNIT_SPELLCAST_SUCCEEDED(unit, spell)
		if spell == goo then
			self:Sync("GooBall")
		end
	end
end
```

**Comment by johngitboy** [comment]
@Guzzone yes!I found this problem in Warmane Lordaeron. Can it be repaired on client side?

**Comment by Guzzone** [comment]
It most definitely can, a friend of mine fixed it.
Later today when I get to the PC I can copy the fixed function and paste it here.

**Comment by Guzzone** [comment]
I'm really sorry @johngitboy I completely forgot to get back to you.
I'm not entirely sure how the fix works as I didn't work on it directly.

Look for the file in dbm called _putricide.lua_ (DBM-Icecrown/ThePlagueworks/Putricide.lua)
I think you just need to put `"CHAT_MSG_RAID_BOSS_EMOTE"` inside this function:
```
mod:RegisterEvents(
	"SPELL_CAST_START",
	"SPELL_CAST_SUCCESS",
	"SPELL_AURA_APPLIED",
	"SPELL_AURA_APPLIED_DOSE",
	"SPELL_AURA_REFRESH",
	"SPELL_AURA_REMOVED",
	"UNIT_HEALTH",
	"CHAT_MSG_RAID_BOSS_EMOTE"
)
```

and this is the actual fix:

```
-- Malleable goo fix
function mod:CHAT_MSG_RAID_BOSS_EMOTE(msg)
	msg = plaintext(msg)

	if msg == gooEmote then
		warnMalleableGoo:Show()
		specWarnMalleableGooCast:Show()
		if mod:IsDifficulty("heroic10") or mod:IsDifficulty("heroic25") then
			timerMalleableGooCD:Start(20)
		else
			timerMalleableGooCD:Start()
		end
	end
end
```

You can download the file from here: https://www.dropbox.com/s/3nrvddlr6xp7pq5/Putricide.lua?dl=1

**Comment by johngitboy** [comment]
@Guzzone ty for your reply，I'll try your solution code when I have time.
and by the way, I checked and found that the spell id of MalleableGoo in trinitycore sourcecode is 70852, not 72615, 72295, 74280, 74281. Maybe it's the reason why DBM can't handle it correctly. 

thanks.

**Comment by Guzzone** [comment]
Well that's the issue, there is no SPELL_CAST_SUCCESS or UNIT_SPELLCAST_SUCCEEDED in the combat log for malleable goo, regardless of Spell Id which is what all the addons used to read.
Even the ones who normally read the emotes such as BigWigs used to read the combat log because of the emote.

I checked before I posted to see if the SpellId was somehow wrong because fixing that is very very easy (you just need to add a number into a function), turns out it just doesn't appear in the combat log.

**Comment by johngitboy** [comment]
Got it. It's interest that the retail client run correctly(the malleable goo's always a big problem in our raid run ) without any spell cast log or event hooked . How it worked? It must be some commands from the server, isn't it ?

**Comment by Guzzone** [comment]
I think the retail server had working malleable goo in combat log since you can see videos of the fight where you see both the emote and dbm's announcement and the dbm everyone uses is the retail version.

**Comment by johngitboy** [comment]
That's not what I mean.(Im not good at english) 
I mean retail server and retail client worked ok:malleable goo is in combat log and dbm can catch that.But why TC server and retail client worked good without malleable goo event in combat log. Client recved the event from TC server but didnt output to combat log?

**Comment by Guzzone** [comment]
Which TC server works with Malleable Goo, are you sure it doesn't appear in combat log?
The missing Goo in combat log might simply be a dalaran-wow issue so it might work well in some other TC servers I guess.
